### PR TITLE
Wdb 18/registration form

### PR DESCRIPTION
**Pull request by tmaciulis22**
- Created Registration From
- Also, improved login form layout spacing

image

### CODE REVIEW

File: frontend/src/components/RegistrationForm/RegistrationForm.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+  Paper,
+  TextField,
+  Typography
+} from '@material-ui/core';
+import visibilityIcon from '../../assets/generic/visibility.svg';
+import visibilityOffIcon from '../../assets/generic/visibility-off.svg';
+import productIcon from "../../assets/generic/product-icon.svg"
+import { emailRegex, passwordRegex, nameRegex } from "../../constants/regex"
+
+export default function RegistrationForm(props) {
```
Comment (kibartas) [inline frontend/src/components/RegistrationForm/RegistrationForm.js:17]: not gonna destructure this?

File: frontend/src/components/RegistrationForm/RegistrationForm.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  const [confirmPassword, setConfirmPassword] = useState("")
+  const [showPassword, setShowPassword] = useState(false)
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false)
+  const [showError, setShowError] = useState({ 
+    errorFirstName: false,
+    errorLastName: false,
+    errorEmail: false,
+    errorPassword: false,
+    errorConfirmPassword: false
+  })
```
Comment (kibartas) [inline frontend/src/components/RegistrationForm/RegistrationForm.js:31]: ugh, where are all the semicolons, bro? Can you run prettier + eslint manually please?

File: frontend/src/components/RegistrationForm/RegistrationForm.js [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+  }
+
+  const handleConfirmPasswordChange = (event) => {
+    setConfirmPassword(event.target.value)
+    setShowError({ ...showError, errorConfirmPassword: false })
+  }
+
+  const handleClickShowPassword = () => setShowPassword(!showPassword)
+  const handleClickShowConfirmPassword = () => setShowConfirmPassword(!showConfirmPassword)
+  const handleMouseDownPassword = (event) => event.preventDefault()
```
Comment (kibartas) [inline frontend/src/components/RegistrationForm/RegistrationForm.js]: what does this do

File: frontend/src/components/RegistrationForm/RegistrationForm.js [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+  return (
+    <Paper elevation={3}>
+      <CardContent
+        direction="column"
+        align="center"
+        justify="center"
+      >
+        <img
+          src={productIcon}
```
Comment (kibartas) [inline frontend/src/components/RegistrationForm/RegistrationForm.js:95]: is this the same as in LoginPage? If so, can you just create a separate reusable component to be used here, please?

File: frontend/src/components/RegistrationForm/RegistrationForm.js [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+                    variant="contained"
+                    color="primary"
+                    fullWidth
+                  >
+                    Register
+                  </Button>
+                </Grid>
+                <Grid item xs={6}>
+                  <Button variant="outlined" fullWidth href="/">
+                    Login
```
Comment (kibartas) [inline frontend/src/components/RegistrationForm/RegistrationForm.js]: wouldn't 'Back to Login' be more appropriate here?

File: frontend/src/components/RegistrationForm/RegistrationForm.js (hunk 1 above)
Comment (tmaciulis22) [reply frontend/src/components/RegistrationForm/RegistrationForm.js:17]: No, next PR

File: frontend/src/components/RegistrationForm/RegistrationForm.js (hunk 2 above)
Comment (tmaciulis22) [reply frontend/src/components/RegistrationForm/RegistrationForm.js:31]: Ugh, do we really need semicolons?

File: frontend/src/components/RegistrationForm/RegistrationForm.js (hunk 4 above)
Comment (tmaciulis22) [reply frontend/src/components/RegistrationForm/RegistrationForm.js:95]: Don't see a reason here, project timeline is short, I think we should not spend time for tiny details as this

File: frontend/src/components/RegistrationForm/RegistrationForm.js (hunk 5 above)
Comment (tmaciulis22) [reply frontend/src/components/RegistrationForm/RegistrationForm.js]: Ok, chief

File: frontend/src/components/RegistrationForm/RegistrationForm.js (hunk 1 above)
Comment (tmaciulis22) [reply frontend/src/components/RegistrationForm/RegistrationForm.js:17]: In logout button's PR to be precise

File: frontend/src/components/RegistrationForm/RegistrationForm.js (hunk 2 above)
Comment (tmaciulis22) [reply frontend/src/components/RegistrationForm/RegistrationForm.js:31]: Ran Prettier in logout-button branch

**Review by kibartas — APPROVED** [review]

File: frontend/src/components/RegistrationForm/RegistrationForm.js (hunk 3 above)
Comment (tmaciulis22) [reply frontend/src/components/RegistrationForm/RegistrationForm.js]: Deleted this nonsense, sorry I was tired, didn't notice it

**Review by AArlauskas — APPROVED** [review]

**Pull request merged**